<launch>

    <arg name="how_many_robot" value="3"/>
    <param name="explore-lite" value="false"/>
    <arg name="cluster_radius" value="2.7"/>
    <param name="cluster_radius" value="$(arg cluster_radius)"/>
    <arg name="world" value="map23"/>
    <param name="world" value="$(arg world)"/>

    <!-- world -->
    <include file="$(find test_unknown_rendezvous)/launch/world_sdf.launch">
    </include>

    <!-- robot 1 -->
    <include file="$(find test_unknown_rendezvous)/launch/one_robot.launch">
        <arg name="robot_number" value="1"/>
        <arg name="pose_x"       value="-3"/>
        <arg name="pose_y"       value="9"/>
        <arg name="pose_z"       value="0"/>
        <arg name="slam"         value="true"/>
        <arg name="formation"    value="true"/>
        <arg name="exploration"  value="true"/>
        <arg name="blobbing"     value="true"/>
    </include>

    <!-- robot 2 -->
    <include file="$(find test_unknown_rendezvous)/launch/one_robot.launch">
        <arg name="robot_number" value="2"/>
        <arg name="pose_x"       value="20"/>
        <arg name="pose_y"       value="10"/>
        <arg name="pose_z"       value="0"/>
        <arg name="slam"         value="true"/>
        <arg name="formation"    value="true"/>
        <arg name="exploration"  value="true"/>
        <arg name="blobbing"     value="true"/>
    </include>
    
    <!-- robot 3 -->
    <include file="$(find test_unknown_rendezvous)/launch/one_robot.launch">
        <arg name="robot_number" value="3"/>
        <arg name="pose_x"       value="4"/>
        <arg name="pose_y"       value="33"/>
        <arg name="pose_z"       value="0"/>
        <arg name="slam"         value="true"/>
        <arg name="formation"    value="true"/>
        <arg name="exploration"  value="true"/>
        <arg name="blobbing"     value="true"/>
    </include>



    <!-- all_positions -->
    <node 
        name   = "pos_aggregator"
        pkg    = "test_unknown_rendezvous"
        type   = "pos_aggregator.py"
        args   = "$(arg how_many_robot)"
        output = "screen"
    />
    
    <!-- cluster -->
    <node 
        name   = "cluster"
        pkg    = "test_unknown_rendezvous"
        type   = "cluster_controller.py"
        args   = "$(arg cluster_radius) $(arg how_many_robot)"
        output = "screen"
    />

    <!-- RVIZ -->
    <include file="$(find test_unknown_rendezvous)/launch/rviz.launch">
        <arg name="rviz_file" value="test_8"/>
    </include> 
</launch>